.github/workflows/ci-tests.yml

```diff
@@ -131,13 +131,13 @@ jobs:
     - name: "prepare coverage"
       if: ${{ matrix.coverage }}
       run: |
-        mv .coverage .coverage${{ strategy.job-index }}
+        mv .coverage ci-test-coverage${{ strategy.job-index }}
```

Comment:
There is a breaking-change released for [actions/upload-artifact v4.4.0](https://github.com/actions/upload-artifact/releases/tag/v4.4.0) which now won't upload hidden files by default. This was breaking the code coverage reporting.

This new default breaking behaviour can be overridden, but we choose to avoid it altogether.

Note that this change in behaviour is for cyber security reasons i.e., to avoid accidentally uploading credentials into artifacts.